cloudformationTemplates/startChatContactAPI/README.md

```diff
@@ -1,83 +1,173 @@
-
-    ```js
-    connect.ChatInterface.initiateChat({
-      name: customerName,
-      username: username,
-      region: ${region},
-      apiGatewayEndpoint: "https://${apiId}.execute-api.${region}.amazonaws.com/Prod",
-      contactAttributes: JSON.stringify({
-        "customerName": customerName
-      }),
-      contactFlowId: "${contactFlowId}",
-      instanceId: "${instanceId}"
-    },successHandler, failureHandler)
+        $(function () {
+            $('#contactDetails').submit(function (e) {
+                e.preventDefault();
+                var contactFlowId = "xxxx1111-xxxx-xxxx-xxxx-xxxxxxxxxxxx"; // TODO: Fill in
+                var instanceId = "xxxx2222-xxxx-xxxx-xxxx-xxxxxxxxxxxx"; // TODO: Fill in
+                var apiGatewayEndpoint = "https://${apiId}.execute-api.${region}.amazonaws.com/Prod"; // TODO: Fill in with the API Gateway endpoint created by your CloudFormation template
+                var regionName = "us-west-2" // TODO: Fill in
+
+                customerName = $('#firstName').val();
+                if (!customerName) {
+                    alert('you must enter a name & username');
+                    document.getElementById("contactDetails").reset();
+                } else {
+            
+                    console.log("this is the first name: " + customerName);
+                    document.getElementById("contactDetails").reset();
+
+                    connect.ChatInterface.initiateChat({
+                        name: customerName,
+                        region: regionName,
+                        apiGatewayEndpoint: apiGatewayEndpoint,
+                        contactAttributes: JSON.stringify({
+                            "customerName": customerName
+                        }),
+                        contactFlowId: contactFlowId,
+                        instanceId: instanceId
+                    },successHandler, failureHandler);
+
+                }
+            });
+        });
+
+        function successHandler(chatSession) {
+            console.log("success!");
+            $('#section-chat').fadeIn(400);
+
+            chatSession.onChatDisconnected(function(data) {
+                $('#section-chat').hide('slide');
+            });
+        }
+
+        function failureHandler(error) {
+            console.log("There was an error: ");
+            console.log(error);
+        }
+    </script>
     ```
+5. Type your name and click on start chat. And then Go to your connect contact center https://<YourContactCenterName>.awsapps.com/connect/home and click on "Test Chat". Change your agent status as available to test the chat workflow
```

Comment:
I would not direct them to the test chat experience. That is to test the customer side. You should instead direct them to the CCP v2 so they can answer the chat that is in queue.

Reply:
Yup. Agreed